### PR TITLE
Update orttraining release pipelines to use torch 1.11.0

### PR DESCRIPTION
**Pull request by baijumeswani**
This pull request introduces pipelines for release with the following config:
1. orttraining built with cuda 11.5 and tested against torch 1.11.0.
2. orttraining built with cuda 11.3 and tested against torch 1.11.0.
3. orttraining built with cuda 10.2 and tested against torch 1.11.0.

Other pipelines that used other torch versions have been removed. Moving forward, the orttraining packaging pipelines will only be tested against the latest stable torch version. The release pipelines will support as many versions of cuda as required by PTCA images.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-cuda113.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ stages:
 - template: templates/py-packaging-training-cuda-stage.yml
   parameters:
     build_py_parameters: --enable_training --update --build
-    torch_version: '1.10.0'
+    torch_version: '1.11.0'
     opset_version: '14'
     cuda_version: '11.3'
```
Comment (ytaous) [inline tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-cuda113.yml:9]: do we still keep this if we have 11.5?

**Review by ytaous — DISMISSED** [review]

**Comment by askhade** [comment]
why do we need both 11.3 and 11.5 versions for cuda?

File: tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_training_cuda10_2 [hunk 2]
```diff
@@ -181,10 +174,9 @@ ENV SSL_CERT_FILE=/opt/_internal/certs.pem
 CMD ["/bin/bash"]
 
 #Build manylinux2014 docker image end
-
-ARG PYTHON_VERSION=3.6
+ARG PYTHON_VERSION=3.9
 ARG TORCH_VERSION=1.11.0
-ARG OPSET_VERSION=12
+ARG OPSET_VERSION=14
```
Comment (askhade) [inline tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_training_cuda10_2:179]: I understand this is related to the ONNX opset version but what does this really control? Is this related to what we test in the pipeline?

File: tools/ci_build/github/linux/docker/scripts/training/ortmodule/stage1/requirements_torch1.11.0_cu10.2.txt [hunk 3]
```diff
@@ -1,5 +1,4 @@
---pre
--f https://download.pytorch.org/whl/torch_stable.html
+--extra-index-url https://download.pytorch.org/whl/cu102
```
Comment (snnn) [inline tools/ci_build/github/linux/docker/scripts/training/ortmodule/stage1/requirements_torch1.11.0_cu10.2.txt:1]: Our security team has concerns on uses of "--extra-index-url". It has risks of supply chain attacks. If pytorch team publishes a fake numpy package to https://download.pytorch.org/, then we will get numpy from pytorch instead of pypi.  See https://docs.microsoft.com/en-us/nuget/concepts/security-best-practices#nuget-feeds for more information. Though the doc was written for nuget, it applies to python too.  But, to address the concern we will need to setup an Azure Devops feed first. We can do it in another PR.

**Comment by snnn** [comment]
Would please do me a favor? Rename all the requirements txt files to "requirements.txt"(without extra words). You can create a separated folder for each of them. It helps security scanning.

File: tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-cuda102.yml [hunk 4]
```diff
@@ -1,15 +1,14 @@
...
-    enable_windows_cpu: false
-    enable_windows_gpu: false
-    enable_mac_cpu: false
-    enable_linux_arm: false
+    build_py_parameters: --enable_training --update --build 
+    torch_version: '1.11.0'
+    opset_version: '14'
+    cuda_version: '10.2'
+    gcc_version: 8
+    cmake_cuda_architectures: 35;37;50;52;60;61;70
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-cuda102.yml:11]: "cmake_cuda_architectures", this is also something what you may want to discuss with the PTC team. Anything not in this list is not guaranteed to work(it might be an onnx runtime unique problem).

Comment (askhade) [reply tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-cuda102.yml:11]: +1 let's check the requirement with PTCA. If they don't have any specific requirements, I suggest removing 35 and adding 75 and 80 as these capabilities apply to Turing and Ampere architectures.

File: tools/ci_build/github/linux/docker/scripts/training/ortmodule/stage1/requirements_torch1.11.0_cu10.2.txt (hunk 3 above)
Comment (baijumeswani) [reply tools/ci_build/github/linux/docker/scripts/training/ortmodule/stage1/requirements_torch1.11.0_cu10.2.txt]: I changed this because PyTorch has updated their installation guidelines to use "--extra-index-url". Will update this once we have a clear path forward. Thanks for sharing the information though.

**Review by snnn — DISMISSED** [review]
Thank you!

**Review by snnn — APPROVED** [review]
